main/src/library/Reducible.flix

```diff
@@ -84,52 +84,62 @@ pub lawless class Reducible[t: Type -> Type] {
     ///
     /// Returns the first element of `t`.
     ///
-    pub def head(t: t[a]): a = ???
+    pub def head(t: t[a]): a =
```

Comment:
This is obviously a tad slow, so the Nel instance should over `head` specifically to get O(1).